bingo.js

```diff
@@ -517,14 +553,33 @@ function toggleColourSymbols(value)
 {
 	COLOURSYMBOLS = !COLOURSYMBOLS;
 	updateColourSymbols();
-	pushNewLocalSetting(COLOUR_SYMBOLS_SETTING_NAME, COLOURSYMBOLS);	
+	pushNewLocalSetting(COLOUR_SYMBOLS_SETTING_NAME, COLOURSYMBOLS);
 }
 
-function pushNewUrl()
+function generateNewUrlParam()
 {
 	var hidden = HIDDEN ? "1" : "0";
 	var streamerMode = STREAMER_MODE ? "1" : "0";
-	window.history.pushState('', "Sheet", "?s=" + DIFFICULTY + "-" + hidden + "-" + streamerMode + "-" + VERSION.id + "_" + SEED);
+	return DIFFICULTY + "-" + hidden + "-" + streamerMode + "-" + VERSION.id + "_" + SEED;
+}
+
+function generateNewUrlParamWithProgress(squareProgress)
+{
+	if (squareProgress === undefined) {
+		if (gup(URL_PARAM_PROGRESS)) {
+			squareProgress = generateSquareProgress();
+		} else {
+			return "?s=" + generateNewUrlParam();
+		}
+	}	
+	
+	return "?s=" + generateNewUrlParam() + "&progress=" + JSON.stringify(squareProgress);
+}
+
+function pushNewUrl(url)
+{
+	if (url === undefined) url = "?s=" + generateNewUrlParam();
```

Comment:
Not that there is a consistent code style, but one-line `if`s without braces aren't used. It would be better to try to keep the style more consistent.